internal/manifests/configmap/configmap.go

```diff
@@ -0,0 +1,135 @@
+package configmap
+
+import (
+	"context"
+	"reflect"
+
+	"github.com/ViaQ/logerr/kverrors"
+	"github.com/ViaQ/logerr/log"
+	"github.com/openshift/elasticsearch-operator/internal/manifests/status"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/util/retry"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// CompareFunc is the type for functions that compare two configmaps.
+// Return true if two configmaps are equal.
+type CompareFunc func(current, desired *corev1.ConfigMap) bool
+
+// MutateFunc is the type for functions that mutate the current configmap
+// by applying the values from the desired configmap.
+type MutateFunc func(current, desired *corev1.ConfigMap)
+
+// Get returns the k8s configmap for the given object key or an error.
+func Get(ctx context.Context, c client.Client, key client.ObjectKey) (*corev1.ConfigMap, error) {
+	cm := New(key.Name, key.Namespace, nil, nil)
+
+	if err := c.Get(ctx, key, cm); err != nil {
+		return nil, kverrors.Wrap(err, "failed to get configmap",
+			"name", cm.Name,
+			"namespace", cm.Namespace,
+		)
+	}
+
+	return cm, nil
+}
+
+// Create will create the given configmap on the api server or return an error on failure
+func Create(ctx context.Context, c client.Client, cm *corev1.ConfigMap) (status.OperationResultType, error) {
```

Comment:
this can likely stay package scoped -- most the time we call `CreateOrUpdate`

Reply:
This is used by `elasticsearch/dashboards.go` as per not using the create or update pattern but a first delete and then create. I am not 100% why dashboards behave differently, but I was encountering strage errors with `CreateOrUpdate` because dashboards reconcile a configmap in a different namespace namely `openshift-config-managed`.